Add initial support for auto-instrumentation (#464)
open-telemetry/opentelemetry-operator: README.md

EOF
```

### OpenTelemetry auto-instrumentation injection

> The operator can inject and configure OpenTelemetry auto-instrumentation libraries. At this moment, the operator can inject only

Reviewer: We do not have forced line wraps in this file, as editors can do it as well. While I don't have a particular preference, I do prefer the same style to be used consistently.

OpenTelemetry [Java auto-instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation).

The injection of the Java agent can be enabled by adding an annotation to the namespace, so that all pods within that
namespace will get the instrumentation, or by adding the annotation to individual PodSpec objects, available as part
of Deployment, Statefulset, and other resources.

```console
instrumentation.opentelemetry.io/inject-java: "true"
```

The value can be 
* `false` - do not inject

> * `true` - inject

Reviewer: What happens in this case? If it follows the same resolution mechanism as the sidecar injection, it might make sense to mention here.

* `java-instrumentation` - name of `Instrumentation` CR instance.

> In the addition to the annotation the following `CR` has to be created. The `Instrumentation`

Reviewer: ```suggestion
In addition to the annotation, the following `CR` has to be created. The `Instrumentation` resource
```

provides configuration for OpenTelemetry SDK and auto-instrumentation.

```yaml
kubectl apply -f - <<EOF
apiVersion: opentelemetry.io/v1alpha1
kind: Instrumentation
metadata:
  name: java-instrumentation
spec:
  exporter:
    endpoint: http://otel-collector:4318
  java:
    image: ghcr.io/pavolloffay/otel-javaagent:1.5.3 # <1>
EOF
```

1. Container image with [OpenTelemetry Java auto-instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation). 

>    The image has to contain Java agent JAR `/javaagent.jar` and the JAR is copied to a shared volume mounted to the application container.

Reviewer: ```suggestion
   The image must contain the Java agent JAR at `/javaagent.jar`, and the operator will copy the JAR to a shared volume mounted to the application pod.
```

@jstickler would be proud of me, reducing the usage of the passive voice :)
